extras/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/KinesisBackedBlockRDD.scala

```diff
@@ -39,8 +39,15 @@ case class SequenceNumberRange(
 private[kinesis]
 case class SequenceNumberRanges(ranges: Array[SequenceNumberRange]) {
   def isEmpty(): Boolean = ranges.isEmpty
+
   def nonEmpty(): Boolean = ranges.nonEmpty
+
   override def toString(): String = ranges.mkString("SequenceNumberRanges(", ", ", ")")
+
+  override def equals(obj: Any): Boolean = {
```

Comment:
This `equals` seems only for testing. However, it doesn't follow the contract of `equals`, such as handling `null`. In addition, `equals` and `hashCode` should be implemented together.

Actually, we should not use an `Array` as a field of case class. How about changing it to `Seq`?